### PR TITLE
send amount to refunds

### PR DESCRIPTION
**Pull request by ivanfer**
missed refunds back in https://github.com/activemerchant/active_merchant/pull/1887. Interestingly enough refunds still work without the amount category but this would allow partial refunds.

@aprofeit @girasquid

### CODE REVIEW

**Comment by j-mutter** [comment]
Remote test? Otherwise LGTM

**Comment by aprofeit** [comment]
Is there docs for this somewhere? How did you know this needed to change?

**Comment by ivanfer** [comment]
The `commit` method had the `add_amount` method originally, the refactor I did in #1887 left our refunds. So this restores previous behaviour.

Credentials look expired on this gateway - I get `Access refusé ou site/lang/clé invalide` on the remote requests. so I found some on this page: http://www1.paybox.com/wp-content/uploads/2014/02/ParametresTestPaybox_V6.2_FR.pdf , under `4.2.4 Tests Paybox System et Paybox Direct non 3D-Secure`, I'll add these

**Comment by ivanfer** [comment]
Adding the credentials showed that the refunds method was not working. It's passing now so the fix works. 

I'll take suggestions on how to fix this encoding error on this test:
```
Failure: test_unsuccessful_purchase(RemotePayboxDirectTest)
test/remote/gateways/remote_paybox_direct_test.rb:31:in `test_unsuccessful_purchase'
     28:   def test_unsuccessful_purchase
     29:     assert response = @gateway.purchase(@amount, @declined_card, @options)
     30:     assert_failure response
  => 31:     assert_equal "PAYBOX : Numéro de porteur invalide", response.message
     32:   end
     33:
     34:   def test_authorize_and_capture
<"PAYBOX : Numéro de porteur invalide">(UTF-8) expected but was
<"PAYBOX : Num\xE9ro de porteur invalide">(ASCII-8BIT)
```

**Comment by ivanfer** [comment]
I fixed the test and now we have 100% remote coverage
```
vagrant@vagrant:~/src/active_merchant$ bet test/remote/gateways/remote_paybox_direct_test.rb
Loaded suite test/remote/gateways/remote_paybox_direct_test
Started
....test/remote/gateways/remote_paybox_direct_test.rb:66:in `test_purchase_and_partial_credit': Support for using credit to refund existing transactions is deprecated and will be removed from a future release of ActiveMerchant. Please use the refund method instead.
....

Finished in 17.0197768 seconds.
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
8 tests, 35 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
0.47 tests/s, 2.06 assertions/s
```

@j-mutter @aprofeit pls2review

File: test/remote/gateways/remote_paybox_direct_test.rb [hunk 1]
```diff
@@ -28,7 +28,7 @@ def test_successful_purchase
   def test_unsuccessful_purchase
     assert response = @gateway.purchase(@amount, @declined_card, @options)
     assert_failure response
-    assert_equal "PAYBOX : Numéro de porteur invalide", response.message
+    assert_equal "PAYBOX : Num\xE9ro de porteur invalide".force_encoding('ASCII-8BIT'), response.message
```
Comment (j-mutter) [inline test/remote/gateways/remote_paybox_direct_test.rb]: trololol encodings

**Comment by j-mutter** [comment]
No remote test for refunds?

**Comment by aprofeit** [comment]
Remote refund test please otherwise this'll break again and no one will know.

**Comment by j-mutter** [comment]
Turns out the remotes for this gateway are still using the deprecated `credit` method. @ivanfer is going to change it to refund.

@girasquid @ntalbott - Thoughts on doing a purge of all `credit` methods in the next major version bump?

**Comment by ivanfer** [comment]
Added explicit remote refunds tests. I'll leave the credit - type tests for when we cull them.

**Comment by ntalbott** [comment]
I have a nice branch with most of the deprecated stuff cleaned up. Hope to get back to it soon.

File: test/remote/gateways/remote_paybox_direct_test.rb [hunk 2]
```diff
@@ -68,14 +67,46 @@ def test_purchase_and_partial_credit
     assert_success credit
   end
   
+  def test_successful_refund
```
Comment (j-mutter) [inline test/remote/gateways/remote_paybox_direct_test.rb]: :heart_eyes:

**Comment by j-mutter** [comment]
:ship:

**Comment by aprofeit** [comment]
Just needs a squash, but :+1:

**Pull request closed without merging**